src/resolve.ts

```diff
@@ -122,6 +128,14 @@ export function resolveModuleURL<O extends ResolveOptions>(
   // Absolute path to file (fast path)
   if (absolutePath) {
     try {
+      if (!options?.preserveSymlinks) {
+        const real = realpathSync(absolutePath);
```

Comment:
Since we are paying cost of one stat (on original path), wondering if it would be faster if we do `realpathSync` only if `lstatSync.isSymbolicLink()` is true. (`lstatSync` behaves same as `statSync` if path is normal file)

Reply:
Updated 🚀